ports/espressif/bindings/espnow/ESPNow.c

```diff
@@ -0,0 +1,943 @@
+    }
+    return true;
+}
+
+// Update the cached peer count in self->peer_count;
+// The peer_count ignores broadcast and multicast addresses and is used for the
+// send() logic and is updated from add_peer(), mod_peer() and del_peer().
+STATIC void _update_peer_count(void) {
+    esp_espnow_obj_t *self = _get_singleton_initialised();
+
+    esp_now_peer_info_t peer = {0};
+    bool from_head = true;
+    int count = 0;
+    // esp_now_fetch_peer() skips over any broadcast or multicast addresses
+    while (esp_now_fetch_peer(from_head, &peer) == ESP_OK) {
+        from_head = false;
+        if (++count >= ESP_NOW_MAX_TOTAL_PEER_NUM) {
+            break;          // Should not happen
+        }
+    }
+    self->peer_count = count;
+}
+
+// ESPNow.add_peer(peer_mac, [lmk, [channel, [ifidx, [encrypt]]]]) or
+// ESPNow.add_peer(peer_mac, [lmk=b'0123456789abcdef'|b''|None|False],
+//          [channel=1..11|0], [ifidx=0|1], [encrypt=True|False])
+// Positional args set to None will be left at defaults.
+// Raise OSError if ESPNow.init() has not been called.
+// Raise ValueError if mac or LMK are not bytes-like objects or wrong length.
+// Raise TypeError if invalid keyword args or too many positional args.
+// Return None.
+STATIC mp_obj_t espnow_add_peer(
+    size_t n_args, const mp_obj_t *args, mp_map_t *kw_args) {
+
+    esp_now_peer_info_t peer = {0};
+    memcpy(peer.peer_addr, _get_peer(args[1]), ESP_NOW_ETH_ALEN);
+    _update_peer_info(&peer, n_args - 2, args + 2, kw_args);
+
+    check_esp_err(esp_now_add_peer(&peer));
+    _update_peer_count();
+
+    return mp_const_none;
+}
+STATIC MP_DEFINE_CONST_FUN_OBJ_KW(espnow_add_peer_obj, 2, espnow_add_peer);
+
+// ESPNow.del_peer(peer_mac): Unregister peer_mac.
+// Raise OSError if ESPNow.init() has not been called.
+// Raise ValueError if peer is not a bytes-like objects or wrong length.
+// Return None.
+STATIC mp_obj_t espnow_del_peer(mp_obj_t _, mp_obj_t peer) {
+    uint8_t peer_addr[ESP_NOW_ETH_ALEN];
+    memcpy(peer_addr, _get_peer(peer), ESP_NOW_ETH_ALEN);
+
+    check_esp_err(esp_now_del_peer(peer_addr));
+    _update_peer_count();
+
+    return mp_const_none;
+}
+STATIC MP_DEFINE_CONST_FUN_OBJ_2(espnow_del_peer_obj, espnow_del_peer);
+
+// Convert a peer_info struct to python tuple
+// Used by espnow_get_peer() and espnow_get_peers()
+static mp_obj_t _peer_info_to_tuple(const esp_now_peer_info_t *peer) {
+    return NEW_TUPLE(
+        mp_obj_new_bytes(peer->peer_addr, MP_ARRAY_SIZE(peer->peer_addr)),
+        mp_obj_new_bytes(peer->lmk, MP_ARRAY_SIZE(peer->lmk)),
+        mp_obj_new_int(peer->channel),
+        mp_obj_new_int(peer->ifidx),
+        (peer->encrypt) ? mp_const_true : mp_const_false);
+}
+
+// ESPNow.get_peers(): Fetch peer_info records for all registered ESPNow peers.
+// Raise OSError if ESPNow.init() has not been called.
+// Return a tuple of tuples:
+//     ((peer_addr, lmk, channel, ifidx, encrypt),
+//      (peer_addr, lmk, channel, ifidx, encrypt), ...)
```

Comment:
inline doc

Reply:
we'd normally make something like this a read-only property instead of a `get_()` method.